client/app/components/parameters.html

```diff
@@ -19,7 +19,11 @@
           <option ng-repeat="option in extractEnumOptions(param.enumOptions)" value="{{option}}">{{option}}</option>
         </select>
       </span>
+      <span ng-switch-when="query" ng-controller="QueryBasedParameterController">
+        <select ng-model="param.value" class="form-control" ng-options="queryResult for queryResult in queryResults">
+        </select>
+      </span>
       <input ng-switch-default type="{{param.type}}" class="form-control" ng-model="param.ngModel">
     </span>
-  </div>
+  </
```

Comment:
Because it works, tests might not help here either. But there might be a linter for HTML we can add to the build process.